### DISCUSSION TITLE
[Question] How to select GPU?

### DISCUSSION THREAD

**Discussion by santoshbs** [explosion/spaCy#6510] [Help: Coding & Implementations] [answered]
Is there a way I can specify spacy to use a particular gpu? I have spacy currently running on gpu 0 and would like to run it on gpu 2 instead.

**Comment by santoshbs** [comment]
Figured it out.

```
import os
os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"
os.environ["CUDA_VISIBLE_DEVICES"] = "3"
```

```
import spacy
spacy.require_gpu()
```

**Comment by adrianeboyd** [comment, marked as answer]
For anyone coming across this issue in the future, you can also set the GPU with:

```python
spacy.require_gpu(gpu_id=3)
```

This should be supported for `spacy>=2.2.2`.
